site/app/controllers/OfficeHoursQueueController.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+namespace app\controllers;
+
+use app\libraries\Core;
+use app\libraries\response\Response;
+use app\libraries\response\WebResponse;
+use app\libraries\response\JsonResponse;
+use app\libraries\response\RedirectResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use app\models\OfficeHoursQueueStudent;
+use app\libraries\routers\AccessControl;
+
+/**
+ * Class OfficeHoursQueueController
+ *
+ */
+class OfficeHoursQueueController extends AbstractController {
+
+    public function __construct(Core $core) {
+        parent::__construct($core);
+    }
+
+    /**
+    * @param
+    * @Route("/{_semester}/{_course}/office_hours_queue")
+    * @return Response
+    */
+    public function showQueue(){
+        if(!$this->core->getConfig()->isQueueEnabled()){
+            return Response::RedirectOnlyResponse(
+                new RedirectResponse($this->core->buildCourseUrl(['home']))
+            );
+        }
+        if(!$this->core->getUser()->accessGrading()){
+            $oh_queue = $this->core->getQueries()->getQueueByUser($this->core->getUser()->getId());
+            return Response::WebOnlyResponse(
+                new WebResponse(
+                    'OfficeHoursQueue',
+                    'showQueueStudent',
+                    $oh_queue,
+                    $this->core->getConfig()->getCourse()
+                    )
+                );
+            }
+            else{
+                $oh_queue = $this->core->getQueries()->getInstructorQueue();
+                return Response::WebOnlyResponse(
+                    new WebResponse(
+                        'OfficeHoursQueue',
+                        'showQueueInstructor',
+                        $oh_queue,
+                        $this->core->getConfig()->getCourse()
+                        )
+                    );
+                }
+            }
```

Comment:
The indentation here is still not right here as the parentheses and brackets do not line up with what opens them (e.g. the parenthesis on line 57 is for the function, not the else). Here is the indentation I'd like to see:
```
    public function showQueue(){
        if(!$this->core->getConfig()->isQueueEnabled()){
            return Response::RedirectOnlyResponse(
                new RedirectResponse($this->core->buildCourseUrl(['home']))
            );
        }
        if(!$this->core->getUser()->accessGrading()){
            $oh_queue = $this->core->getQueries()->getQueueByUser($this->core->getUser()->getId());
            return Response::WebOnlyResponse(
                new WebResponse(
                    'OfficeHoursQueue',
                    'showQueueStudent',
                    $oh_queue,
                    $this->core->getConfig()->getCourse()
                )
            );
        }
        else {
            $oh_queue = $this->core->getQueries()->getInstructorQueue();
            return Response::WebOnlyResponse(
                new WebResponse(
                    'OfficeHoursQueue',
                    'showQueueInstructor',
                    $oh_queue,
                    $this->core->getConfig()->getCourse()
                )
            );
        }
    }
```